samu.c

```diff
@@ -79,6 +79,59 @@ warnflag(const char *flag)
 		fatal("unknown warning flag '%s'", flag);
 }
 
+void
+parseenvargs(char *s)
+{
+	char *copy;
+	char *p;
+	int argc;
+	enum { maxArgs = 64 };
+	char *env_argv[maxArgs];
+	char **argv = &env_argv[0];
+	char *arg, *end;
+	long num;
+		
+	if (s == NULL) {
+		return;
+	}
+
+	copy = xstrdup(s);
+	
+	argc = 1;
+	argv[0] = NULL;
+	
+	p = strtok(copy, " ");
+	while (p && argc < maxArgs - 1) {
+		argv[argc++] = p;
+		p = strtok(NULL, " ");
+	}
+	argv[argc] = NULL;
+
+	ARGBEGIN {
+	case '-':
+		arg = EARGF(usage());
+		if (strcmp(arg, "verbose") == 0) {
+			buildopts.verbose = true;
+		} else {
+			fatal("invalid long argument in SAMUFLAGS");
+		}
+		break;
+	case 'j':
+		num = strtol(EARGF(usage()), &end, 10);
```

Comment:
Maybe you could separate `-j` handling into a helper function `jobsflag` to make sure it is handled the same way as if it were passed on the command-line (to prevent them from getting out of sync if one changes).